### PR TITLE
feat: add OBJECTSTORE_PART_SIZE and OBJECTSTORE_CONCURENCY env vars

### PR DESCRIPTION
**Pull request by voroyam**

### CODE REVIEW

File: ENVIRONMENT.md [hunk 1]
```diff
@@ -212,6 +212,10 @@
   Secret key for the objectstore (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/files/external_storage/s3_compatible_object_storage_as_primary.html)).
 - `OWNCLOUD_OBJECTSTORE_VERSION=2006-03-01` \
   Objectstore version to use (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/files/external_storage/s3_compatible_object_storage_as_primary.html)).
+- `OWNCLOUD_OBJECTSTORE_PART_SIZE=5242880` \
+  part size, in bytes, to use when doing a multipart upload (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/files/external_storage/s3_compatible_object_storage_as_primary.html#configuration)).
```
Comment (xoxys) [inline ENVIRONMENT.md:216]:
```suggestion
  part size, in bytes, to use when doing a multipart upload (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/files/external_storage/s3_compatible_object_storage_as_primary.html)).
```

File: ENVIRONMENT.md [hunk 2]
```diff
@@ -212,6 +212,10 @@
...
+- `OWNCLOUD_OBJECTSTORE_CONCURRENCY=5` \
+  maximum number of concurrent UploadPart operations allowed during the multipart upload (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/files/external_storage/s3_compatible_object_storage_as_primary.html#configuration)).
```
Comment (xoxys) [inline ENVIRONMENT.md:218]:
```suggestion
  maximum number of concurrent UploadPart operations allowed during the multipart upload (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/files/external_storage/s3_compatible_object_storage_as_primary.html)).  
```

**Review by xoxys — CHANGES_REQUESTED** [review]
Just a small one to use the same URLs that are already used for the objectstore vars.

File: ENVIRONMENT.md [hunk 3]
```diff
@@ -212,6 +212,10 @@
   Secret key for the objectstore (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/files/external_storage/s3_compatible_object_storage_as_primary.html)).
 - `OWNCLOUD_OBJECTSTORE_VERSION=2006-03-01` \
   Objectstore version to use (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/files/external_storage/s3_compatible_object_storage_as_primary.html)).
+- `OWNCLOUD_OBJECTSTORE_PART_SIZE=5242880` \
+  part size, in bytes, to use when doing a multipart upload (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/files/external_storage/s3_compatible_object_storage_as_primary.html#configuration)).
+- `OWNCLOUD_OBJECTSTORE_CONCURRENCY=5` \
```
Comment (d7oc) [inline ENVIRONMENT.md:217]:
```suggestion
- `OWNCLOUD_OBJECTSTORE_CONCURRENCY=3` \
```

Default concurrency is 3 we should leave the default.

File: latest/overlay/etc/entrypoint.d/40-objectstorage.sh [hunk 4]
```diff
@@ -27,4 +27,10 @@ declare -x OWNCLOUD_OBJECTSTORE_ENDPOINT
 declare -x OWNCLOUD_OBJECTSTORE_PATHSTYLE
 [[ -z "${OWNCLOUD_OBJECTSTORE_PATHSTYLE}" ]] && OWNCLOUD_OBJECTSTORE_PATHSTYLE="false"
 
+declare -x OWNCLOUD_OBJECTSTORE_PART_SIZE
+[[ -z "${OWNCLOUD_OBJECTSTORE_PART_SIZE}" ]] && OWNCLOUD_OBJECTSTORE_PART_SIZE="5242880"
+
+declare -x OWNCLOUD_OBJECTSTORE_CONCURRENCY
+[[ -z "${OWNCLOUD_OBJECTSTORE_CONCURRENCY}" ]] && OWNCLOUD_OBJECTSTORE_CONCURRENCY="5"
```
Comment (d7oc) [inline latest/overlay/etc/entrypoint.d/40-objectstorage.sh:34]:
```suggestion
[[ -z "${OWNCLOUD_OBJECTSTORE_CONCURRENCY}" ]] && OWNCLOUD_OBJECTSTORE_CONCURRENCY="3"
```

see above.

File: v20.04/overlay/etc/entrypoint.d/40-objectstorage.sh [hunk 5]
```diff
@@ -27,4 +27,10 @@ declare -x OWNCLOUD_OBJECTSTORE_ENDPOINT
 declare -x OWNCLOUD_OBJECTSTORE_PATHSTYLE
 [[ -z "${OWNCLOUD_OBJECTSTORE_PATHSTYLE}" ]] && OWNCLOUD_OBJECTSTORE_PATHSTYLE="false"
 
+declare -x OWNCLOUD_OBJECTSTORE_PART_SIZE
+[[ -z "${OWNCLOUD_OBJECTSTORE_PART_SIZE}" ]] && OWNCLOUD_OBJECTSTORE_PART_SIZE="5242880"
+
+declare -x OWNCLOUD_OBJECTSTORE_CONCURRENCY
+[[ -z "${OWNCLOUD_OBJECTSTORE_CONCURRENCY}" ]] && OWNCLOUD_OBJECTSTORE_CONCURRENCY="5"
```
Comment (d7oc) [inline v20.04/overlay/etc/entrypoint.d/40-objectstorage.sh:34]:
```suggestion
[[ -z "${OWNCLOUD_OBJECTSTORE_CONCURRENCY}" ]] && OWNCLOUD_OBJECTSTORE_CONCURRENCY="3"
```

see above.

**Review by d7oc — APPROVED** [review]
The default `part_size` is unknown to me. Should be checked with engineering. Maybe
@jvillafanez knowns.

Rest LGTM, aside my comments.

**Comment by jvillafanez** [comment]
> The default part_size is unknown to me. Should be checked with engineering. Maybe
@jvillafanez knowns.

Maybe we should clarify it applies to uploads between ownCloud and S3, not between the client and ownCloud.
I don't think the "multipart" is relevant, and it could also confuse the user because it could be interpreted as if the multipart upload happens between the client and ownCloud. Since the upload happens between ownCloud and S3, whether the upload is multipart or not doesn't seem important to the user; he just need to know that the file will be split in chunks of that size and the chunks will be uploaded to S3.

**Comment by xoxys** [comment]
That's just a copy/paste + back ref to doc.owncloud.com. If that requires adjustments, it should be adjusted there first. The remaining question is still what's the default part size used by the library?

**Comment by jvillafanez** [comment]
5 MB according to https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/s3-multipart-upload.html
>  part_size
    (int, default: int(5242880)) Part size, in bytes, to use when doing a multipart upload. This must between 5 MB and 5 GB, inclusive

**Comment by voroyam** [comment]
@xoxys leave the `#configuration` at the end of the link or remove it?

**Comment by xoxys** [comment]
Remove it please.

**Comment by voroyam** [comment]
doc issue to change the concurrency parameter there to 3 instead of 5
https://github.com/owncloud/docs/issues/4816

**Comment by voroyam** [comment]
doc issue to change the description of concurrency:
https://github.com/owncloud/docs/issues/4817

**Review by xoxys — APPROVED** [review]

File: ENVIRONMENT.md [hunk 6]
```diff
@@ -212,6 +212,10 @@
   Secret key for the objectstore (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/files/external_storage/s3_compatible_object_storage_as_primary.html)).
 - `OWNCLOUD_OBJECTSTORE_VERSION=2006-03-01` \
   Objectstore version to use (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/files/external_storage/s3_compatible_object_storage_as_primary.html)).
+- `OWNCLOUD_OBJECTSTORE_PART_SIZE=5242880` \
+  part size, in bytes, applies to uploads between ownCloud and S3 (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/files/external_storage/s3_compatible_object_storage_as_primary.html)).
+- `OWNCLOUD_OBJECTSTORE_CONCURRENCY=3` \
+  maximum number of concurrent UploadPart operations allowed during the multipart upload (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/files/external_storage/s3_compatible_object_storage_as_primary.html)).
```
Comment (xoxys) [inline ENVIRONMENT.md:218]: Please start with an uppercase.

File: ENVIRONMENT.md [hunk 7]
```diff
@@ -212,6 +212,10 @@
   Secret key for the objectstore (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/files/external_storage/s3_compatible_object_storage_as_primary.html)).
 - `OWNCLOUD_OBJECTSTORE_VERSION=2006-03-01` \
   Objectstore version to use (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/files/external_storage/s3_compatible_object_storage_as_primary.html)).
+- `OWNCLOUD_OBJECTSTORE_PART_SIZE=5242880` \
+  part size, in bytes, applies to uploads between ownCloud and S3 (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/files/external_storage/s3_compatible_object_storage_as_primary.html)).
```
Comment (xoxys) [inline ENVIRONMENT.md:216]: Please start with an uppercase.

**Review by xoxys — CHANGES_REQUESTED** [review]

**Pull request merged**